### ISSUE TITLE
NET-1802 FN S3956: Properties declared via primary constructor not detected

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#6416]
This code is detected by S3956 complaining about a property exposing a `List<int>`:

```cs
public record R
{
    public List<int> Property { get; init; }
//         ^^^^^^^^^ Noncompliant {{Refactor this property to use a generic collection designed for inheritance.}}
}
```

The equivalent shorthand for is not recognized properly:

```cs
public record R2(List<int> Property);  // FN. Should report on "Property"
```

Test case added: #6417

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1802](https://sonarsource.atlassian.net/browse/NET-1802)

[NET-1802]: https://sonarsource.atlassian.net/browse/NET-1802?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
